jdaviz/app.py

```diff
@@ -109,72 +109,76 @@ def to_unit(self, data, cid, values, original_units, target_units):
                 spec = data.get_object(cls=Spectrum1D)
 
             except RuntimeError:
-                eqv = []
+                data = data.get_object(cls=NDDataArray)
+                spec = Spectrum1D(flux=data.data * u.Unit(original_units))
+            return self._flux_conversion(spec, values, original_units, target_units)
+        else:  # spectral axis
+            return self._spectral_axis_conversion(values, original_units, target_units)
+
+    def _flux_conversion(self, spec, values, original_units, target_units):
+        # If there are only two values, this is likely the limits being converted, so then
+        # in case we need to use the spectral density equivalency, we need to provide only
+        # to spectral axis values. If there is only one value
+        if not np.isscalar(values) and len(values) == 2:
+            spectral_values = spec.spectral_axis[0]
+        else:
+            spectral_values = spec.spectral_axis
+
+        # Ensure a spectrum passed through Spectral Extraction plugin
+        if '_pixel_scale_factor' in spec.meta and len(values) != 2:
+
+            # Data item in data collection does not update from conversion/translation.
+            # App wide original data units are used for conversion, original and
+            # target_units dictate the conversion to take place.
+
+            if (u.sr in u.Unit(original_units).bases) and \
+                    (u.sr not in u.Unit(target_units).bases):
+                # Surface Brightness -> Flux
+                eqv = [(u.MJy / u.sr,
+                        u.MJy,
+                        lambda x: (x * np.array(spec.meta['_pixel_scale_factor'])),
```

Comment:
Should `spec.meta['_pixel_scale_factor']` be `spec.meta.get('_pixel_scale_factor', 1)` or similar, in case it hasn't been defined?